src/ucs/config/parser.c

```diff
@@ -1090,14 +1257,44 @@ static inline int ucs_config_is_table_field(const ucs_config_field_t *field)
     return (field->parser.read == ucs_config_sscanf_table);
 }
 
+static inline int ucs_config_is_key_value_field(const ucs_config_field_t *field)
+{
+    return (field->parser.read == ucs_config_sscanf_key_value);
+}
+
+/**
+ * Allocates multi-line documentation text string, caller is responsible for
+ * freeing the memory.
+ */
+static char *ucs_config_get_doc(const ucs_config_field_t *field)
+{
+    const ucs_config_key_value_param_t *param;
+    const ucs_config_key_field_t *cfg_key;
+    ucs_string_buffer_t strb;
+
+    ucs_string_buffer_init(&strb);
+    ucs_string_buffer_appendf(&strb, "%s", field->doc);
+
+    if (ucs_config_is_key_value_field(field)) {
```

Comment:
can't it be part of ucs_config_help_key_value?

Reply:
It can, but help is also used to warn about config errors, so the consequence would be:
When user makes an error in config, we show him a help message, that would be much larger, e.g.
```
 Allowed keys:
    front      front temperature
    rear       rear temperature
    syntax:    default value for all keys and/or comma-separated list of key:value pairs, where key is one of allowed keys and value is: unsigned integer
``` 
Please confirm if this is desired behavior

Reply:
why do we print "allowed keys" and not the list of keys?
anyway we should not have special cases for config types in the general parsing code

Reply:
Doc:
```
Specifies car temperatures.
- front : front temperature 
- rear  : rear temperature
```

Syntax:
```
Comma-separated list of value or key:value pairs, where key is one of {front, rear} and 
value is unsigned integer. A value without a key is the default.
```

Reply:
Ok!
Another agreement is to add new `doc` function pointer to parser, so that we don't need to handle specific case here

Reply:
Done.
I just slightly changed the formatting to resemble existing configs:
```
#
# MMAP hook mode
#  none   - Don't set mmap hooks.
#  reloc  - Use ELF relocation table to set hooks.
#  bistro - Use binary instrumentation to set hooks.

# Cuda memory hook modes. A combination of:
#  none   - Don't set Cuda hooks.
#  reloc  - Use ELF relocation table to set hooks. In this mode, if any

# Module unload prevention mode
#  lazy - use RTLD_LAZY flag to add reference to module.
#  now  - use RTLD_NOW flag to add reference to module.

# Communication scheme in RNDV protocol.
#  get_zcopy - use get_zcopy scheme in RNDV protocol.
#  put_zcopy - use put_zcopy scheme in RNDV protocol.
```
So now it looks like this:
```
#
# Temperature
#  front     - front temperature
#  rear      - rear temperature
#
# syntax:    comma-separated list of value or key:value pairs, where key is one of [front,rear] and value is: unsigned integer. A value without a key is the default.
#
UCX_TEMP=front:17,rear:16
```